### PR TITLE
Route errors handling data portion

### PR DESCRIPTION
**Pull request by msecret**
This addresses all the data that's needed to get error handling working for route errors. The UI should be able to use this data to show the errors to the user.

### CODE REVIEW

File: static_src/actions/route_actions.js [hunk 1]
```diff
@@ -108,5 +115,13 @@ export default {
       routeGuid,
       route
     });
+  },
+
+  error(routeGuid, err) {
```
Comment (jcscottiii) [inline static_src/actions/route_actions.js]: is this a general error for the routes API? trying to figure out why we need both this one and the one for the route create case.

File: static_src/actions/route_actions.js [hunk 2]
```diff
@@ -108,5 +115,13 @@ export default {
...
+    AppDispatcher.handleServerAction({
+      type: routeActionTypes.ROUTE_ERROR,
+      routeGuid,
```
Comment (jcscottiii) [inline static_src/actions/route_actions.js]: any reason why we don't use key-value mapping for the `routeGuid` too?

File: static_src/constants.js [hunk 3]
```diff
@@ -172,7 +174,8 @@ const routeActionTypes = keymirror({
   ROUTE_UPDATED: null,
   ROUTES_FOR_APP_FETCH: null,
   ROUTES_FOR_APP_RECEIVED: null,
-  ROUTE_TOGGLE_EDIT: null
+  ROUTE_TOGGLE_EDIT: null,
```
Comment (jcscottiii) [inline static_src/constants.js]: add comment

File: static_src/constants.js [hunk 4]
```diff
@@ -172,7 +174,8 @@ const routeActionTypes = keymirror({
...
+  ROUTE_ERROR: null
```
Comment (jcscottiii) [inline static_src/constants.js]: add comment

File: static_src/test/unit/stores/route_store.spec.js [hunk 5]
```diff
@@ -25,11 +26,13 @@ describe('RouteStore', function() {
   describe('constructor()', function() {
     it('should start data as empty array', function() {
       expect(RouteStore.getAll()).toBeEmptyArray();
+      expect(RouteStore.error).toEqual(null);
     });
   });
 
   describe('routeActionTypes.ROUTE_APP_ASSOCIATED', function () {
-    it('should add app_guid to the route object and set editing to false', function () {
+    it('should add app_guid to the route object and set editing, erro to false',
```
Comment (jcscottiii) [inline static_src/test/unit/stores/route_store.spec.js]: typo `erro` -> `error`

File: static_src/actions/route_actions.js (hunk 1 above)
Comment (jcscottiii) [inline static_src/actions/route_actions.js]: update, i read the rest, i can see the need for the two. just put the comment below in `constants.js`

**Comment by jcscottiii** [comment]
few comments. LGTM!

File: static_src/actions/route_actions.js (hunk 2 above)
Comment (msecret) [inline static_src/actions/route_actions.js]: It's an es6 thing haha

Comment (jcscottiii) [inline static_src/actions/route_actions.js]: nvm this. @msecret explained this.

**Pull request merged**